pkg/encoding/v1/encoding.go

```diff
@@ -179,10 +179,55 @@ type ImageRef string
 
 // FIXME: implement ImageRef unmarshalling
 
+type Path string
```

Comment:
How about we move the type declaration to object.go and do something like `type Path object.Path` here?